website/content/en/highlights/2024-05-06-0-38-0-upgrade-guide.md

```diff
@@ -0,0 +1,57 @@
+---
+date: "2024-05-06"
+title: "0.38 Upgrade Guide"
+description: "An upgrade guide that addresses breaking changes in 0.38.0"
+authors: ["jszwedko"]
+release: "0.38.0"
+hide_on_release_notes: false
+badges:
+  type: breaking change
+---
+
+Vector's 0.38.0 release includes a **breaking change**:
+
+1. [Vector now requires mandatory environment variables](#strict-env-vars)
+
+We cover them below to help you upgrade quickly:
+
+and a **deprecation**:
+
+1. [Deprecation of undefined environment variable warnings](#strict-env-vars)
+
+## Upgrade guide
+
+### Breaking Change
+
+#### Vector now requires mandatory environment variables {#strict-env-vars}
+
+Vector has the ability to [interpolate environment variables into its
+configuration](/docs/reference/configuration/#environment-variables) as a means of templating
+configuration. Now, if an environment variable is undefined, and no default is set, Vector will
+output an error and stop. To set a default, use the `-` interpolation syntax.
+
+If you want to have Vector start-up even if an environment variable is undefined you can provide
+a default like `${FOO-bar}` to default `FOO` to `bar` if it is unset.
+
+The reason for this change is that users often miss the undefined variable warning and are confused
+by Vector not behaving as the expected. Certain characters need to be escaped in regex capture
```

Comment:
might not be an issue but thought it slightly odd to reference users in this perspective, when users are the ones reading the release notes. We could rephrase maybe to remove the users language?

Reply:
This was actually copied from the previous release note, but I'm not sure what subject to use instead 🤔 Did you have one in mind?

Reply:
Ah 👍  Non blocking , just an idea off the cuff:

```
The motivation for this change is that the undefined variable warning is often overlooked which causes confusion as to why Vector is not behaving as expected.
```

Reply:
"causes confusion" _to whom?_ 😂  We seem to tend to use `users` frequently in the docs and release notes.